### PR TITLE
Gdplus article cover

### PR DESCRIPTION
**Pull request by DriftingSands**
Refresh the page 3 times to use up the 'free articles'
New article cover will appear which will not allow users to continue reading.

Fix #187

Test URLs:
- Before: https://main--helix-sportsmagazine--headwirecom.hlx.page/content-v2/golf-news-tours/_default/article/2023/6/open-championship-2023-ra-chairman-defends-proper-name-golfs-oldest-major
- After: https://gdplus-article-cover--helix-sportsmagazine--headwirecom.hlx.page/content-v2/golf-news-tours/_default/article/2023/6/open-championship-2023-ra-chairman-defends-proper-name-golfs-oldest-major

- Before: https://main--helix-sportsmagazine--headwirecom.hlx.page/content-v2/golf-news-tours/_default/article/2023/4/how-course-designer-beau-welling-balances-a-life-of-golf-and-curling-the-fringe
- After: https://gdplus-article-cover--helix-sportsmagazine--headwirecom.hlx.page/content-v2/golf-news-tours/_default/article/2023/4/how-course-designer-beau-welling-balances-a-life-of-golf-and-curling-the-fringe

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -644,3 +650,41 @@ export const premiumArticleBanner = (leftoverArticles = 0) => {
     </div>
   `;
 };
+/**
+ * Generates HTML for the premium article blocker.
+ * @param {block} Block where the selector exists.
+ * @param {Selector} Selector for article body that should be covered.
+ */
+export const generateArticleBlocker = (block, selector) => {
+  if (Number(window.sessionStorage.freeArticles) > 0) {
```
Comment (icaraps) [inline scripts/scripts.js:659]: I think we can remove that log

File: scripts/scripts.js [hunk 2]
```diff
@@ -621,7 +621,13 @@ window.store = new (class {
  * Generates HTML for the premium article banner.
  * @param {Number} Number of leftover articles to compare to.
  */
-export const premiumArticleBanner = (leftoverArticles = 0) => {
+export const premiumArticleBanner = (customLeftoverArticles = null) => {
+  let leftoverArticles = customLeftoverArticles;
+  if (typeof customLeftoverArticles !== 'number') {
+    leftoverArticles = Number(window.sessionStorage.freeArticles || 3);
+  }
+  window.sessionStorage.freeArticles = Math.max(leftoverArticles - 1, 0);
```
Comment (icaraps) [inline scripts/scripts.js:629]: sessionStorage can be disabled by the user (by default disabled in Chrome incognito mode AFAIR) so maybe we should rely on something else eg use the `window.name` trick WDYT ?

**Review by icaraps — CHANGES_REQUESTED** [review]

File: scripts/scripts.js (hunk 1 above)
Comment (DriftingSands) [reply scripts/scripts.js:663]: the log was removed

File: scripts/scripts.js (hunk 2 above)
Comment (DriftingSands) [reply scripts/scripts.js]: I changed to use window.name

**Review by icaraps — CHANGES_REQUESTED** [review]
@DriftingSands  Instead of icons/GD_Plus.png, can you publish the png in drive and then use `createOptimizedPicture()` to generate the logo ?

In the icons folder, you usually have SVG icons only.

**Pull request merged**